domain/consensus/processes/pruningmanager/pruningmanager.go

```diff
@@ -551,3 +584,33 @@ func (pm *pruningManager) updatePruningPointUTXOSet() error {
 	log.Debugf("Finishing updating the pruning point UTXO set")
 	return pm.pruningStore.FinishUpdatingPruningPointUTXOSet(pm.databaseContext)
 }
+
+func (pm *pruningManager) PruneAllBlocksBelow(pruningPointHash *externalapi.DomainHash) error {
+	onEnd := logger.LogAndMeasureExecutionTime(log, "PruneAllBlocksBelow")
+	defer onEnd()
+
+	iterator, err := pm.blocksStore.AllBlockHashesIterator(pm.databaseContext)
+	if err != nil {
+		return err
+	}
+
+	for ok := iterator.First(); ok; ok = iterator.Next() {
+		blockHash, err := iterator.Get()
+		if err != nil {
+			return err
+		}
+		isInPastOfPruningPoint, err := pm.dagTopologyManager.IsDescendantOf(blockHash, pruningPointHash)
```

Comment:
nit, I'd prefer if we just called `IsAncestorOf(pruningPointHash, blockHash)` and completely remove `IsDescendantOf`

Reply:
Fun fact: I was the only user of IsDescenadntOf, so removed it.